### ISSUE TITLE
Implement remove

### ISSUE THREAD

**Issue by ldeluigi** [BBVA/mercury-settrie#7]
Allow users to remove sets from the SetTrie

```py
s = SetTrie()
s.remove(id)

# These methods would also be nice to have:
s.remove_supersets({1, 2, 3})
s.remove_subsets({1, 2, 3})
```

**Comment by sbasaldua** [comment]
Hi ldeluigi,

If we implemented .remove(), the other two would be very straightforward since .find() is very fast. The problem is .remove() would not be implementable efficiently without some architecture changes:

1. The BinaryTree is only designed for adding, it is a vector and the index to each element is permanent for the life of the tree. If we start creating unused elements, we would have to track them with some other structure to be able to reuse them.
2. The .remove() itself would not be ideal unless the structure is modified with an index to the parent. The tree is designed to travel down, not up.
3. The StringName that stores the strings for a given hash does not have a counter of the number of times the same string is used. To remove unused strings, that would have to be implemented.

Most of the team is still on vacation until next week, give us a few more days to decide, but it looks like a lot of work.

Thank you very much for your interest.

**Comment by sbasaldua** [comment]
Hi again,

Good news!

After implementing the iterators, we are much closer to implementing delete(). Stay tuned!

**Comment by sbasaldua** [comment]
Good news, again!

We have implemented remove() in version 1.4.5.

Thank you for your suggestions.

**Comment by ldeluigi** [comment]
Great! Awesome work! Thank you
